bin/mark_changepoints_in_json

```diff
@@ -46,49 +148,59 @@ def main(in_files, window_size):
                     outliers = krun_data[filename]['all_outliers'][bench][index]
                 else:
                     outliers = list()
-                c_points_all, means = get_changepoints(cpt, p_exec[:], outliers)
-                # The last datum is always a changepoint, so we ignore it.
-                c_points = c_points_all[:-1]
-                changepoints[bench].append(c_points)
-                changepoint_means[bench].append(means)
-                classifications[bench].append(get_classification(p_exec, c_points,
-                                                                 means, window_size))
+                segments = get_segments(cpt, p_exec[:], steady_state, outliers,
+                                        use_snr_penalty=penalty)
+                changepoints[bench].append(segments.changepoints)
+                changepoint_means[bench].append(segments.means)
+                classifications[bench].append(segments.get_classification())
         krun_data[filename]['changepoints'] = changepoints
-        krun_data[filename]['classifications'] = classifications
         krun_data[filename]['changepoint_means'] = changepoint_means
+        krun_data[filename]['classifications'] = classifications
+        krun_data[filename]['steady_state_expected'] = steady_state
         new_filename = create_output_filename(filename)
         print 'Writing out: %s' % new_filename
         write_krun_results_file(krun_data[filename], new_filename)
 
 
-def get_changepoints(cpt, p_exec, outliers):
+def _get_snr_penalty(data):
+    penalty = numpy.mean(data) / numpy.std(data)
+    if numpy.isnan(penalty) or numpy.isinf(penalty):
+        raise ValueError()
+    else:
+        return penalty
+
+
+def get_segments(cpt, p_exec, steady_state, outliers, use_snr_penalty=False):
+    data = p_exec[:]
+    length = len(p_exec)  # Will change when we remove outliers.
     indices = sorted(outliers, reverse=True)
     for index in indices:
         del p_exec[index]
     measurements = rpy2.robjects.FloatVector(p_exec)
-    changepoints = cpt.cpt_meanvar(measurements, method='PELT')
+    if use_snr_penalty:  # User has asked for SNR penalty.
+        try:  # Check it doesn't return NaN or Inf.
+            penalty = _get_snr_penalty(p_exec[-steady_state:])
+            changepoints = cpt.cpt_meanvar(measurements, method='PELT',
+                                           penalty='Manual', pen_value=penalty)
+        except ValueError:  # Use default penalty, 'MBIC'.
+            changepoints = cpt.cpt_meanvar(measurements, method='PELT')
```

Comment:
No, 'MBIC' would be an argument to 'penalty'. 'PELT' means we want 1 *or more* changepoints.